.clang-tidy

```diff
@@ -1,7 +1,8 @@
 ---
-# TODO: Enable as many useful checks as possible.
-# E.g. set to '*', see which of those aren't useful, and disable just those.
-Checks: '-*,readability-redundant-declaration'
+# TODO(jkff): Re-enable google-readability-casting: it can be annoying, but it
+# can catch some nasty bugs. E.g. C-style casts happily cast a pointer to the
+# wrong type.
+Checks: '*,-*-magic-numbers,-hicpp-*,-cppcoreguidelines-*,-fuchsia-*,-clion-*,-cert-*,-readability-named-parameter,-llvm-header-guard,-google-readability-todo,-misc-unused-parameters,-*-braces-around-statements,-google-readability-casting,-readability-else-after-return,-modernize-use-auto,-modernize-deprecated-headers,-llvm-include-order'
```

Comment:
It looks like an include-order rule snuck in here; do we know it's consistent with clang-format?

Reply:
This says `-llvm-include-order`, disabling it - I disabled it because it turned out that it's, in fact, inconsistent with clang-format.

Reply:
Ah, I see.

Can we make this a whitelist instead of a blacklist?  Since we're running different versions of clang-tidy, presumably they have different checks included in `*`, and we only want to run the checks common to both of them?